polygeist/tools/cgeist/driver.cc

```diff
@@ -361,13 +361,17 @@ static int optimize(mlir::MLIRContext &Ctx,
   mlir::PassManager PM(&Ctx);
   enableOptionsPM(PM);
 
-  mlir::OpPassManager &OptPM = PM.nestAny();
   GreedyRewriteConfig CanonicalizerConfig;
   CanonicalizerConfig.maxIterations = CanonicalizeIterations;
 
   if (OptLevel != llvm::OptimizationLevel::O0) {
     PM.addPass(polygeist::createArgumentPromotionPass());
 
+    mlir::OpPassManager &OptPM = PM.nestAny();
+    if (EnableLICM)
+      OptPM.addPass(polygeist::createLICMPass(
+          {options.getCgeistOpts().getRelaxedAliasing()}));
```

Comment:
Maybe the pipeline change should be performed in a different PR to see if this is where this pass should go.